### PR TITLE
Improve dev script

### PR DESCRIPTION
**Pull request by mikkoh**
This PR does two things. Makes `dev` watch `.ts` files also and attempts to improve build speed for `watch`.

So I've seen #235 which is more extensive where it breaks out tests etc.

I'm putting this out there to bring this up as a discussion again.

Even after this PR `build` takes ~2sec to run which is not nice. I'm not actually sure why builds take so long. Might investigate at some point.

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -13,9 +13,7 @@
  * limitations under the License.
  */
 
-const fs = require('fs');
 const path = require('path');
-const rollup = require('rollup');
```
Comment (mikkoh) [inline rollup.config.js]: These weren't actually being used so I 🔥 them.

File: rollup.config.dev.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  const filesToIgnore = IGNORE_WARNINGS[warning.code];
+  if (filesToIgnore && filesToIgnore.indexOf(warning.id) !== -1) {
+    return;
+  }
+
+  warn(warning);
+};
+
+const plugins = [
+  resolve(),
```
Comment (mikkoh) [inline rollup.config.dev.js]: I removed `clean` here in hopes that it would speed up builds.

File: rollup.config.dev.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+};
+
+const onwarn = (warning, warn) => {
+  const filesToIgnore = IGNORE_WARNINGS[warning.code];
+  if (filesToIgnore && filesToIgnore.indexOf(warning.id) !== -1) {
+    return;
+  }
+
+  warn(warning);
+};
```
Comment (mikkoh) [inline rollup.config.dev.js]: I recognize this code could be put in it's module. I can do this in this PR if you all want.

File: rollup.config.js (hunk 1 above)
Comment (cdata) [reply rollup.config.js]: 👍

**Comment by mikkoh** [comment]
I'm cool with `--environment` if everyone else is cool with this. I believe #235 didn't use this.

**Comment by cdata** [comment]
It would make it easier to be specific about what we want in dev vs prod without sharing code and / or duplicating configuration. And, we could have more than just dev and prod scenarios (what about test?).

File: rollup.config.js [hunk 4]
```diff
@@ -13,12 +13,11 @@
...
 const resolve = require('rollup-plugin-node-resolve');
 const cleanup = require('rollup-plugin-cleanup');
 
+const {NODE_ENV} = process.env;
```
Comment (mikkoh) [inline rollup.config.js]: I think using `NODE_ENV` has become a standard so 🤷‍♂️

**Comment by mikkoh** [comment]
I've updated to use `environment` instead of two distinct configs.

File: package.json [hunk 5]
```diff
@@ -31,7 +31,8 @@
   ],
   "scripts": {
     "clean": "rm -rf ./lib ./dist",
-    "build": "tsc && rollup -c",
+    "build": "tsc && rollup -c --environment NODE_ENV:production",
+    "build:dev": "tsc && rollup -c --environment NODE_ENV:development",
```
Comment (cdata) [inline package.json]: 👍

File: rollup.config.js (hunk 4 above)
Comment (cdata) [reply rollup.config.js]: 👍

**Comment by mrdoob** [comment]
Thanks @mikkoh, my dev day today speed up considerably 🙏

**Comment by mikkoh** [comment]
🙌 I might look into it more and see if we can make some more speedups.

**Pull request merged**